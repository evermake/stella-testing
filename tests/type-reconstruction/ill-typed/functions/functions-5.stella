language core;

extend with #type-reconstruction;

fn twice(f : auto) -> auto {
  return fn(n : auto) {
    return (if Nat::iszero(n) then f(n) else f(f(n)))(n)
  }
}

fn main(n : auto) -> auto {
  return twice( fn(x : auto){ return n } )(n)
}