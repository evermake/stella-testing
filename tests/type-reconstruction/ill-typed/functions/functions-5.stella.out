An error occurred during typechecking!
===================
Primary type error:
===================
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
when unifying expected type
  fn (?T13) -> ?T14
against actual type
  Nat
when typechecking the expression
  if Nat::iszero (n) then f (n) else f (f (n))
========================
Alternative type errors:
========================
Type Error Tag: [ERROR_NOT_A_FUNCTION]
when unifying expected type
  fn (?T13) -> ?T14
against actual type
  Nat
when typechecking the expression
  if Nat::iszero (n) then f (n) else f (f (n))